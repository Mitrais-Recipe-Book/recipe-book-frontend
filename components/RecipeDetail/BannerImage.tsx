import { useRouter } from "next/router";
import React, { useEffect, useState } from "react";
import Image from "next/image";

interface ImgInfo {
  id: string | undefined;
  alt: string;
  href?: boolean;
}

export default function BannerImage(props: ImgInfo) {
  const [img, setImg] = useState("");
  const defaultImage = "/images/bibimbap-image.webp";
  const router = useRouter();

  useEffect(() => {
    setImg(`${process.env.API_URL}recipe/${props?.id}/photo`);
  }, [props.id]);

  function pushTo(): undefined {
    router.push(`/recipe/${props.id}`);
    return;
  }
  return props.href ? (
    <Image
      className=" rounded-t cursor-pointer"
      src={img ? img : defaultImage}
      alt={props.alt}
      width={200}
      height={130}
      layout="responsive"
      objectFit="cover"
      onClick={pushTo}
      onErrorCapture={(e) => {
        setImg(defaultImage);
      }}
    />
  ) : (
    <Image
      className="object-cover rounded-t"
      src={img ? img : defaultImage}
      alt={props.alt}
      width={180}
      height={100}
      layout="responsive"
      objectFit="cover"
      onErrorCapture={(e) => {
        setImg(defaultImage);
      }}
    />
  );
}
