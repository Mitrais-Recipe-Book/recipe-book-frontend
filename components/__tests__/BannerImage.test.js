import BannerImage from "../RecipeDetail/BannerImage";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import renderer from "react-test-renderer";

describe("Test BannerImage that have href", () => {
  const props = {
    id: 1,
    alt: "banner-recipe-es-teh-anget",
    href: true,
  };
  beforeAll(() => {
    process.env.API_URL = "https://recipyb-dev.herokuapp.com/api/v1/";
  });

  it("render without crashing", () => {
    render(<BannerImage {...props} />);
    screen.debug();
  });
  it.todo("has href");
  it("match the snapshot", () => {
    const tree = renderer.create(<BannerImage {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("Test BannerImage that do not have href", () => {
  const props = {
    id: 1,
    alt: "banner-recipe-es-teh-anget",
    href: false,
  };
  beforeAll(() => {
    process.env.API_URL = "https://recipyb-dev.herokuapp.com/api/v1/";
  });

  it("render without crashing", () => {
    render(<BannerImage {...props} />);
  });
  it("has alt", () => {
    render(<BannerImage {...props} />);
    const linkElement = screen.getByAltText(/banner-recipe-es-teh-anget/i);
    expect(linkElement).toBeInTheDocument();
  });
  it.todo("do not have href");
  it("match the snapshot", () => {
    const tree = renderer.create(<BannerImage {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
