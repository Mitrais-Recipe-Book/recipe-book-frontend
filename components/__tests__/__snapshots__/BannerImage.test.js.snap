// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test BannerImage that do not have href match the snapshot 1`] = `
<span
  style={
    Object {
      "background": "none",
      "border": 0,
      "boxSizing": "border-box",
      "display": "block",
      "height": "initial",
      "margin": 0,
      "opacity": 1,
      "overflow": "hidden",
      "padding": 0,
      "position": "relative",
      "width": "initial",
    }
  }
>
  <span
    style={
      Object {
        "background": "none",
        "border": 0,
        "boxSizing": "border-box",
        "display": "block",
        "height": "initial",
        "margin": 0,
        "opacity": 1,
        "padding": 0,
        "paddingTop": "55.55555555555556%",
        "width": "initial",
      }
    }
  />
  <img
    alt="banner-recipe-es-teh-anget"
    className="object-cover rounded-t"
    data-nimg="responsive"
    decoding="async"
    onError={[Function]}
    onErrorCapture={[Function]}
    onLoad={[Function]}
    src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
    style={
      Object {
        "border": "none",
        "bottom": 0,
        "boxSizing": "border-box",
        "display": "block",
        "height": 0,
        "left": 0,
        "margin": "auto",
        "maxHeight": "100%",
        "maxWidth": "100%",
        "minHeight": "100%",
        "minWidth": "100%",
        "objectFit": "cover",
        "objectPosition": undefined,
        "padding": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 0,
      }
    }
  />
  <noscript>
    <img
      alt="banner-recipe-es-teh-anget"
      className="object-cover rounded-t"
      data-nimg="responsive"
      decoding="async"
      loading="lazy"
      onErrorCapture={[Function]}
      sizes="100vw"
      src="/_next/image?url=https%3A%2F%2Frecipyb-dev.herokuapp.com%2Fapi%2Fv1%2Frecipe%2F1%2Fphoto&w=3840&q=75"
      srcSet="/_next/image?url=https%3A%2F%2Frecipyb-dev.herokuapp.com%2Fapi%2Fv1%2Frecipe%2F1%2Fphoto&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Frecipyb-dev.herokuapp.com%2Fapi%2Fv1%2Frecipe%2F1%2Fphoto&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Frecipyb-dev.herokuapp.com%2Fapi%2Fv1%2Frecipe%2F1%2Fphoto&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Frecipyb-dev.herokuapp.com%2Fapi%2Fv1%2Frecipe%2F1%2Fphoto&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Frecipyb-dev.herokuapp.com%2Fapi%2Fv1%2Frecipe%2F1%2Fphoto&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Frecipyb-dev.herokuapp.com%2Fapi%2Fv1%2Frecipe%2F1%2Fphoto&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Frecipyb-dev.herokuapp.com%2Fapi%2Fv1%2Frecipe%2F1%2Fphoto&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Frecipyb-dev.herokuapp.com%2Fapi%2Fv1%2Frecipe%2F1%2Fphoto&w=3840&q=75 3840w"
      style={
        Object {
          "border": "none",
          "bottom": 0,
          "boxSizing": "border-box",
          "display": "block",
          "height": 0,
          "left": 0,
          "margin": "auto",
          "maxHeight": "100%",
          "maxWidth": "100%",
          "minHeight": "100%",
          "minWidth": "100%",
          "objectFit": "cover",
          "objectPosition": undefined,
          "padding": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": 0,
        }
      }
    />
  </noscript>
</span>
`;

exports[`Test BannerImage that have href match the snapshot 1`] = `
<span
  style={
    Object {
      "background": "none",
      "border": 0,
      "boxSizing": "border-box",
      "display": "block",
      "height": "initial",
      "margin": 0,
      "opacity": 1,
      "overflow": "hidden",
      "padding": 0,
      "position": "relative",
      "width": "initial",
    }
  }
>
  <span
    style={
      Object {
        "background": "none",
        "border": 0,
        "boxSizing": "border-box",
        "display": "block",
        "height": "initial",
        "margin": 0,
        "opacity": 1,
        "padding": 0,
        "paddingTop": "65%",
        "width": "initial",
      }
    }
  />
  <img
    alt="banner-recipe-es-teh-anget"
    className=" rounded-t cursor-pointer"
    data-nimg="responsive"
    decoding="async"
    onClick={[Function]}
    onError={[Function]}
    onErrorCapture={[Function]}
    onLoad={[Function]}
    src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
    style={
      Object {
        "border": "none",
        "bottom": 0,
        "boxSizing": "border-box",
        "display": "block",
        "height": 0,
        "left": 0,
        "margin": "auto",
        "maxHeight": "100%",
        "maxWidth": "100%",
        "minHeight": "100%",
        "minWidth": "100%",
        "objectFit": "cover",
        "objectPosition": undefined,
        "padding": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 0,
      }
    }
  />
  <noscript>
    <img
      alt="banner-recipe-es-teh-anget"
      className=" rounded-t cursor-pointer"
      data-nimg="responsive"
      decoding="async"
      loading="lazy"
      onClick={[Function]}
      onErrorCapture={[Function]}
      sizes="100vw"
      src="/_next/image?url=%2Fimages%2Fbibimbap-image.webp&w=3840&q=75"
      srcSet="/_next/image?url=%2Fimages%2Fbibimbap-image.webp&w=640&q=75 640w, /_next/image?url=%2Fimages%2Fbibimbap-image.webp&w=750&q=75 750w, /_next/image?url=%2Fimages%2Fbibimbap-image.webp&w=828&q=75 828w, /_next/image?url=%2Fimages%2Fbibimbap-image.webp&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2Fbibimbap-image.webp&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2Fbibimbap-image.webp&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2Fbibimbap-image.webp&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2Fbibimbap-image.webp&w=3840&q=75 3840w"
      style={
        Object {
          "border": "none",
          "bottom": 0,
          "boxSizing": "border-box",
          "display": "block",
          "height": 0,
          "left": 0,
          "margin": "auto",
          "maxHeight": "100%",
          "maxWidth": "100%",
          "minHeight": "100%",
          "minWidth": "100%",
          "objectFit": "cover",
          "objectPosition": undefined,
          "padding": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": 0,
        }
      }
    />
  </noscript>
</span>
`;
